README.md
<p align="center"><img src="screenshots/screenshot.png"></p>

<h2 align="center">Highlight Search Results in the Browser for Anki</h2>

<p align="center">
<a title="Latest (pre-)release" href="https://github.com/glutanimate/highlight-search-results/releases"><img src ="https://img.shields.io/github/release-pre/glutanimate/highlight-search-results.svg?colorB=brightgreen"></a>
<a title="License: GNU AGPLv3" href="https://github.com/glutanimate/highlight-search-results/blob/master/LICENSE"><img  src="https://img.shields.io/badge/license-GNU AGPLv3-green.svg"></a>
<a title="Rate on AnkiWeb" href="https://ankiweb.net/shared/info/225180905"><img src="https://glutanimate.com/logos/ankiweb-rate.svg"></a>
<br>
<a title="Buy me a coffee :)" href="https://ko-fi.com/X8X0L4YV"><img src="https://img.shields.io/badge/ko--fi-contribute-%23579ebd.svg"></a>
<a title="Support me on Patreon :D" href="https://www.patreon.com/bePatron?u=7522179"><img src="https://img.shields.io/badge/patreon-support-%23f96854.svg"></a>
<a title="Follow me on Twitter" href="https://twitter.com/intent/user?screen_name=glutanimate"><img src="https://img.shields.io/twitter/follow/glutanimate.svg"></a>
</p>

> The quickest way to find information on your flashcards

This is an add-on for the spaced-repetition flashcard app [Anki](https://apps.ankiweb.net/) that extends Anki's card browser with the ability to highlight search terms found in notes.

### Table of Contents <!-- omit in toc -->

<!-- MarkdownTOC levels="1,2,3" -->

- [Installation](#installation)
- [Documentation](#documentation)
- [Building](#building)
- [Contributing](#contributing)
- [License and Credits](#license-and-credits)

<!-- /MarkdownTOC -->

<!-- ### Screenshots

![](screenshots/screenshot.png) -->

### Installation

#### AnkiWeb <!-- omit in toc -->

The easiest way to install Highlight Search Results in the Browser is through [AnkiWeb](https://ankiweb.net/shared/info/225180905).

#### Manual installation <!-- omit in toc -->

Please click on the entry corresponding to your Anki version:

<details>

<summary><i>Recent Anki 2.1 versions</i></summary>

<br>

*Note: These instructions only work on Anki 2.1.17 and up. For older Anki releases please see the next section.*

1. Download the latest `.ankiaddon` file from the [releases tab](https://github.com/glutanimate/highlight-search-results/releases) (you might need to click on *Assets* below the description to reveal the download links)
2. Open the folder where your downloads are located and double-click on the downloaded `.ankiaddon` file.
3. Follow the installation prompt and restart Anki if it asks you to

</details>

<details>

<summary><i>Older Anki 2.1 versions</i></summary>

1. Make sure you are [using at least Anki 2.1.10](https://apps.ankiweb.net/#download). Earlier releases (e.g. found in various Linux distros) do not support `.ankiaddon` packages.
2. Download the latest `.ankiaddon` package from the [releases tab](https://github.com/glutanimate/highlight-search-results/releases) (you might need to click on *Assets* below the description to reveal the download links)
3. From Anki's main window, head to *Tools* → *Add-ons*
4. Drag-and-drop the `.ankiaddon` package onto the add-ons list
5. Restart Anki

Video summary:

<img src="https://raw.githubusercontent.com/glutanimate/docs/master/anki/add-ons/media/ankiaddon-installation-macos.gif" width=640>

</details>

### Documentation

For further information on the use of this add-on please check out [the description text](docs/description.md) for AnkiWeb.

### Building

With [Anki add-on builder](https://github.com/glutanimate/anki-addon-builder/) installed:

    git clone https://github.com/glutanimate/highlight-search-results.git
    cd highlight-search-results
    aab build

For more information on the build process please refer to [`aab`'s documentation](https://github.com/glutanimate/anki-addon-builder/#usage).

### Contributing

Contributions are welcome! Please review the [contribution guidelines](./CONTRIBUTING.md) on how to:

- Report issues
- File pull requests
- Support the project as a non-developer

### License and Credits

*Highlight Search Results in the Browser* is *Copyright © 2017-2020 [Aristotelis P.](https://glutanimate.com/) (Glutanimate)*

Highlight Search Results in the Browser is free and open-source software. The add-on code that runs within Anki is released under the GNU AGPLv3 license, extended by a number of additional terms. For more information please see the [LICENSE](https://github.com/glutanimate/highlight-search-results/blob/master/LICENSE) file that accompanied this program.

Please note that this program uses the [Libaddon](https://github.com/glutanimate/anki-libaddon/) library which comes with [its own additional terms extending the GNU AGPLv3 license](https://github.com/glutanimate/anki-libaddon/blob/master/LICENSE). You may only copy, distribute, or modify the present compilation of this program with Libaddon under the combined licensing terms specified by both licenses.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.


----

<b>
<div align="center">The continued development of this add-on is made possible <br>thanks to my <a href="https://www.patreon.com/glutanimate">Patreon</a> and <a href="https://ko-fi.com/X8X0L4YV">Ko-Fi</a> supporters.
<br>You guys rock ❤️ !</div>
</b>

CHANGELOG.md
# Changelog

All notable changes to Highlight Search Results in the Browser will be documented here. You can click on each release number to be directed to a detailed log of all code commits for that particular release. The download links will direct you to the GitHub release page, allowing you to manually install a release if you want.

If you enjoy Highlight Search Results in the Browser, please consider supporting my work on Patreon, or by buying me a cup of coffee :coffee::

<p align="center">
<a href="https://www.patreon.com/glutanimate" rel="nofollow" title="Support me on Patreon 😄"><img src="https://glutanimate.com/logos/patreon_button.svg"></a>      <a href="https://ko-fi.com/X8X0L4YV" rel="nofollow" title="Buy me a coffee 😊"><img src="https://glutanimate.com/logos/kofi_button.svg"></a>
</p>

:heart: My heartfelt thanks goes out to everyone who has supported this add-on through their tips, contributions, or any other means (you know who you are!). All of this would not have been possible without you. Thank you for being awesome!

## [Unreleased]

## [1.0.1] - 2023-10-21

### [Download](https://github.com/glutanimate/highlight-search-results/releases/tag/v1.0.1)

### Added

- Added support for Anki **23.10** and up (thanks to @zjosua)

## [1.0.0] - 2021-03-10

### [Download](https://github.com/glutanimate/highlight-search-results/releases/tag/v1.0.0)

### Added

- Added support for Anki **2.1.41** and up
- Added support for configuring the add-on through Anki's built-in **add-on config window**
- All **hotkeys** are now fully **configurable**

### Fixed

- Fixed selecting next card on Anki 2.1.24 and up (thanks to @ijgnord for the report, #1)

### Changed

- **Rewrote most of the add-on**, focusing on bringing the add-on up to speed with all the latest and greatest in Anki development. This should make the add-on more robust against upcoming changes in Anki.

    Still, Anki has been undergoing a lot of changes under the hood recently, and these tend to break add-ons. So for the best add-on experience, my advice would be to stick to Anki version you are using until Anki prompts you to update.
- Dropped support for Anki 2.0
- Migrated to new repository and build system

## [1.0.0-beta.1] - 2021-03-07

### [Download](https://github.com/glutanimate/highlight-search-results/releases/tag/v1.0.0-beta.1)

### Added

- Added support for Anki **2.1.41** and up
- Added support for configuring the add-on through Anki's built-in **add-on config window**
- All **hotkeys** are now fully **configurable**

### Fixed

- Fixed selecting next card on Anki 2.1.24 and up (thanks to @ijgnord for the report, #1)

### Changed

- **Rewrote most of the add-on**, focusing on bringing the add-on up to speed with all the latest and greatest in Anki development. This should make the add-on more robust against upcoming changes in Anki.

    Still, Anki has been undergoing a lot of changes under the hood recently, and these tend to break add-ons. So for the best add-on experience, my advice would be to stick to Anki version you are using until Anki prompts you to update.
- Dropped support for Anki 2.0
- Migrated to new repository and build system

## [0.2.0] - 2018-02-17

### Added

- Support for Anki 2.1

## 0.1.0 - 2017-08-16

### Added

- Initial release of Highlight Search Results in the Browser

[Unreleased]: https://github.com/glutanimate/highlight-search-results/compare/v1.0.1...HEAD
[1.0.1]: https://github.com/glutanimate/highlight-search-results/compare/v1.0.0...v1.0.1
[1.0.0]: https://github.com/glutanimate/highlight-search-results/compare/v1.0.0-beta.1...v1.0.0
[1.0.0-beta.1]: https://github.com/glutanimate/highlight-search-results/compare/v0.2.0...v1.0.0-beta.1
[0.2.0]: https://github.com/glutanimate/highlight-search-results/releases/tag/v0.2.0


-----

The format of this file is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

src/highlight_search_results/consts.py
# -*- coding: utf-8 -*-

# Highlight Search Results in the Browser Add-on for Anki
#
# Copyright (C) 2017-2020  Aristotelis P. <https://glutanimate.com/>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version, with the additions
# listed at the end of the license file that accompanied this program.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
#
# NOTE: This program is subject to certain additional terms pursuant to
# Section 7 of the GNU Affero General Public License.  You should have
# received a copy of these additional terms immediately following the
# terms and conditions of the GNU Affero General Public License that
# accompanied this program.
#
# If not, please request a copy through one of the means of contact
# listed here: <https://glutanimate.com/contact/>.
#
# Any modifications to this file must keep this entire header intact.

"""
Addon-wide constants
"""

from ._version import __version__

try:
    from .data.patrons import MEMBERS_CREDITED, MEMBERS_TOP
except ImportError:
    MEMBERS_CREDITED = MEMBERS_TOP = ()


__all__ = [
    "ADDON"
]


class ADDON(object):
    """Class storing general add-on properties
    Property names need to be all-uppercase with no leading underscores
    """
    NAME = "Highlight Search Results in the Browser"
    MODULE = "highlight_search_results"
    ID = "225180905"
    VERSION = __version__
    LICENSE = "GNU AGPLv3"
    AUTHORS = (
        {"name": "Aristotelis P. (Glutanimate)", "years": "2017-2020",
         "contact": "Glutanimate"},
    )
    AUTHOR_MAIL = "ankiglutanimate [αt] gmail . com"
    LIBRARIES = ()
    CONTRIBUTORS = ()
    SPONSORS = ()
    MEMBERS_CREDITED = MEMBERS_CREDITED
    MEMBERS_TOP = MEMBERS_TOP
    LINKS = {
        "patreon": "https://www.patreon.com/glutanimate",
        "bepatron": "https://www.patreon.com/bePatron?u=7522179",
        "coffee": "http://ko-fi.com/glutanimate",
        "description": "https://ankiweb.net/shared/info/{}".format(ID),
        "rate": "https://ankiweb.net/shared/review/{}".format(ID),
        "twitter": "https://twitter.com/glutanimate",
        "youtube": "https://www.youtube.com/c/glutanimate",
        "help": ""
    }
